File: stalin_sort.kt
import kotlin.random.Random
fun stalin_sort(inp: ArrayList<Double>): ArrayList<Double> {
    if (inp.isEmpty()) {
        return arrayListOf<Double>()
    }
    var maxV = inp[0]
    var sorted = arrayListOf<Double>()
    for (i in inp) {
        if (i >= maxV) {
            sorted.add(i)
            maxV = i
        }
        else {
            println("$i sent to Gulag")
        }
    }
    return sorted
}

fun main() {
    var a = arrayListOf<Double>()
    for (j in 0..49) {
        var ww = Random.nextDouble(0.0, 100.0)
        a.add(ww)
    }
    println(a)
    println(stalin_sort(a))
}

File: stalin_sort.ts
function stain_sort(inp: Array<number>) {
    if (inp.length == 0) {
        return inp;
    }

    let max_v = inp[0];
    let sorted: Array<number> = [];

    inp.forEach(element => {
        if (element >= max_v) {
            sorted.push(element);
            max_v = element;
        }
        else {
            console.log(`${element} sent to Gulag`)
        }
    });
    return sorted;
}

let arr: Array<number> = [];
let i: number = 0;
while (i < 50) {
    arr.push(100 * Math.random());
    i++;
}
console.log(arr);
console.log(stain_sort(arr));